src/Npgsql/NpgsqlBatch.cs

```diff
@@ -171,4 +171,30 @@ public override Task PrepareAsync(CancellationToken cancellationToken = default)
 
     /// <inheritdoc />
     public override void Cancel() => Command.Cancel();
+
+    /// <inheritdoc />
+    public override void Dispose()
+    {
+        ClearTransaction();
+        if (Command.IsCached && Connection is not null && Connection.CachedBatch is null)
```

Comment:
Should probably rename IsCached to IsCacheable (here and on command) - IIRC the reason we have it is to avoid caching the command/batch when an exception occurs. Otherwise we'd always do it as long as Connection.CachedCommand is null, no?

Reply:
> IIRC the reason we have it is to avoid caching the command/batch when an exception occurs. Otherwise we'd always do it as long as Connection.CachedCommand is null, no?

Not exactly, that's just so we don't attempt to cache a batch/command if it's created by a user via a constructor, and not from `CreateCommand`. But I agree that we should rename it.